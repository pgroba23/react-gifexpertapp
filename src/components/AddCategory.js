import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleChange = ( e ) => {
        setInputValue( e.target.value );
        //console.log('Paso por aca= '+ inputValue);
    };

    const handleSubmit = ( e ) => {
        e.preventDefault();
        
        if( inputValue.trim().length > 2)
        {
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }


    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
          <input 
            type="text"
            value={ inputValue }
            onChange={ handleChange }
          />  
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
