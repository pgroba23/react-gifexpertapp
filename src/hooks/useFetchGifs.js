import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = ( category ) => {
    
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect( () => {
        getGifs( category )
            .then( img => {
                setState({
                    data: img,
                    loading: false,
                })
            })
    }, [ category ]);
    //     //useEffect es un Hook para que cada vez que se renderize la pagina, las funciones como en este caso getGifs
    //     //se ejecuten solo una vez ante cada renderizacion, sino por cada cambio de estado, react generaria un nuevo
    //     //array de Gifs. Los [] son para indicar que no tiene dependencias, si cambia category se vuelve llamar a get..

    return state;

}


