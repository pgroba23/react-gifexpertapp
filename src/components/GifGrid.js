import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    console.log(images);

   
    return (
        <>
            <h2 className="animate__jello">{ category }</h2>
            {
                loading && <p className="animate__flash">Loading..</p>
            }
            <div className="card-grid">
                    {
                        images.map( img => 
                            <GifGridItem 
                                key= { img.id }
                                {...img} //envia el contenido como parametro
                            />
                        )
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}