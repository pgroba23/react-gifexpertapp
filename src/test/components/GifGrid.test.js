import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('Pruebas sobre GifGrid', () => {
    
    const category = 'One Punch';

    test('GifGrid debe hacer match con su snapshot ', () => {

        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ category }/> );
       
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe mostrar items cuando se cargan imagenes useFetchGifs ', () => {
        
        const gifs =[ {
            id: 'ABC',
            title: 'texto',
            url: 'https://localhost/probandoimagen.jpg',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category }/> );
        expect( wrapper ).toMatchSnapshot();

    })
    
    


})

