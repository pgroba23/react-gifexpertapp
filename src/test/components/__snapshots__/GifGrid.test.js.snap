// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGrid GifGrid debe hacer match con su snapshot  1`] = `
<Fragment>
  <h2
    className="animate__jello"
  >
    One Punch
  </h2>
  <p
    className="animate__flash"
  >
    Loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre GifGrid debe mostrar items cuando se cargan imagenes useFetchGifs  1`] = `
<Fragment>
  <h2
    className="animate__jello"
  >
    One Punch
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="texto"
      url="https://localhost/probandoimagen.jpg"
    />
  </div>
</Fragment>
`;
